billing-api/billing/__init__.py

```diff
@@ -56,20 +57,25 @@ def login():
         username=request.json['username'],
         password=request.json['password'])
     response = Response(status=200, content_type='application/json')
-    response.headers['Authorization'] = token
+    response.headers['Authorization'] = token['token']
     return response
 
 
 @app.route('/projects', methods=['GET'])
 @authenticate
-def get_projects(client):
-    tenants = map(lambda tenant: {'id': tenant.to_dict()['id'], 'name': tenant.to_dict()['name']}, client.tenants.list())
+def get_projects(client, user_id):
+    role_map = database.get_user_roles(user_id)
+    tenants = map(lambda tenant: {'id': tenant.to_dict()['id'],
+                                  'name': tenant.to_dict()['name'],
+                                  'roles': role_map[tenant.to_dict()['id']]},
+                  sessions.list_projects(client))
+
     return tenants
 
 
 @app.route('/reports', methods=['GET'])
 @authenticate
-def calculate_cost_by_user(client):
+def calculate_cost_by_user(client, user_id):
```

Comment:
Unsure of if there's a way to remove that parameter in the signature while still passing user_id to the function